pentasharp/Controllers/AddTaxiDriverController.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using pentasharp.Models.Enums;
+using WebApplication1.Filters;
+using pentasharp.Services;
+using pentasharp.Models.TaxiRequest;
+using pentasharp.Interfaces;
+using pentasharp.ViewModel.TaxiModels;
+using pentasharp.Models.Utilities;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace WebApplication1.Controllers
+{
+    [Route("Business/TaxiDriver")]
+    public class AddTaxiDriverController : Controller
+    {
+        private readonly ITaxiService _taxiService;
+        private readonly IDriverService _driverService;
+        private readonly IAuthenticateService _authService;
+
+        public AddTaxiDriverController(
+            ITaxiService taxiService,
+            IDriverService driverService,
+            IAuthenticateService authService)
+        {
+            _taxiService = taxiService;
+            _driverService = driverService;
+            _authService = authService;
+        }
+
+        [HttpPost("AddDriver")]
+        public async Task<IActionResult> AddDriver([FromBody] RegisterDriverRequest model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(new { success = false, message = "Invalid data provided." });
+            }
+
+            try
+            {
+                var newDriver = await _driverService.AddDriverAsync(model);
+                return Ok(new { success = true, message = "Driver added successfully.", driverId = newDriver.UserId });
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                return Unauthorized(new { success = false, message = ex.Message });
```

Comment:
this exception is better suited for an authentication/authorization issue ant not for adding a driver.
Also add loggers.